RHIROS-32 Implement ROS system detail page

kgaikwad opened the pull request:
With this commit, it renders 
- inventory entity details
- ros system information
- system rating related to recommendations
- add empty recommendation App

Tasks:
- [x] render Inventory details
- [x] make ROS system detail API request
- [x] render ROS system details such as Provider, Instance Type, Idling Time, I/O wait
- [x] add recommendation rating component
- [x] rename RosDetailPage to RosSystemDetail
- [x] add recommendations list
- [x] working review comments & refactoring
- [ ] add filtering on description (Note that - created separate sub-task for filtering in https://issues.redhat.com/browse/RHIROS-109)

Adding Screenshot:
![Screenshot from 2021-04-05 20-05-39](https://user-images.githubusercontent.com/6470528/113589769-68d05a80-964f-11eb-9952-f48690cf8ec3.png)


+ adding @karelhala & @terezanovotna  into loop for review

kgaikwad commented on src/Routes/RosDetailsPage/RosDetailsPage.js:
```diff
@@ -1,15 +1,138 @@
+    }
+
+    async fetchRosSystemDetails() {
+        let url = new URL(ROS_API_ROOT + SYSTEMS_API_ROOT + `/${this.state.inventoryId}`,  window.location.origin);
```
@karelhala, 
One query - I tried to make this change using action & reducer so that it will add ROS system info in store. But it wasn't giving call to reducer function for type `FETCH_ROS_SYSTEM_DETAIL_FULFILLED`. 
Apart from [line-81](https://github.com/RedHatInsights/ros-frontend/pull/24/files#diff-fcf73cad52850091e1789c058644cf5a36de60d32f3014517b98385dac732221R81) change, do I need to make any extra change to get it work?

kgaikwad commented on src/store/entityDetailReducer.js:
```diff
@@ -1,64 +1,35 @@
+            {
+                title: 'Recommendations',
+                name: 'ros_recommendations',
+                component: Recommendations
```
@karelhala,
For component `Recommendations`, may I pass rosDetails as props? or I should make a request inside component itself? Any recommended way while adding App components?

karelhala replied:
I think the safest way would be to refetch the data. Actually it might be fruitfull to create new reducer which will consume these data and you can re-use that reducer in `Recommendations` component and in `Inventory` header.

kgaikwad replied:
+1, added separate reducer to fetch ROS system information.

kgaikwad commented:
@karelhala, 
Updated PR with `SystemDetail` App for ROS system detail page.
Also, added recommendations table component. Now working on your review comments & name filtering.

@terezanovotna - I am attaching a screenshot for system detail page. I saw a comment by @karelhala  on [mock-up page](https://marvelapp.com/prototype/9408d9j/screen/77819763) regarding data list instead of table. Please let me know your thoughts on this. I will modify it to data list.

![Screenshot from 2021-04-08 16-18-40](https://user-images.githubusercontent.com/6470528/114014925-c7334e00-9886-11eb-9832-6b8cefcacff4.png)

kgaikwad commented on src/Components/Recommendations.js:
```diff
@@ -0,0 +1,18 @@
+const Recommendations = () => {
+    return (
+        <Suspense fallback="">
+            <span className='sample-component'>Recommendations</span>
```
Query - when to use `React.lazy` and  `asyncComponent` from `Utilities` while importing any component?

kgaikwad commented on src/Routes/RosDetailsPage/RosDetailsPage.js:
```diff
@@ -1,15 +1,138 @@
+                idling_time: idlingTime, io_wait: ioWait,
+                rating,
+                // FIXME: use inventory_id once backend PR merged
+                host_id: inventoryId
```
Fixed

kgaikwad commented on src/Components/SystemDetail/RecommendationsTable.js:
```diff
@@ -0,0 +1,95 @@
+import { EmptyStateDisplay } from '../EmptyStateDisplay/EmptyStateDisplay';
+import { CheckCircleIcon } from '@patternfly/react-icons';
+import { TextContent, Text, TextVariants } from '@patternfly/react-core';
+import successColor from '@patternfly/react-tokens/dist/esm/global_success_color_100';
```
@karelhala -
I would like to know whether this is a correct way to import color for icon or not.

kgaikwad replied:
Modified import statement.

karelhala commented on src/Components/SystemDetail/RecommendationsTable.js:
```diff
@@ -0,0 +1,103 @@
+import { EmptyStateDisplay } from '../EmptyStateDisplay/EmptyStateDisplay';
+import { CheckCircleIcon } from '@patternfly/react-icons';
+import { TextContent, Text, TextVariants } from '@patternfly/react-core';
+import {
+    global_success_color_100 as successColor
+} from '@patternfly/react-tokens';
```
No need to use `react-tokens` value. They tend to bloat the size of app bundle. It is much better to style this trough class in `EmptyStateDisplay` and its icon.

kgaikwad replied:
Done! Removed react-tokens from package.json & applied icon color using CSS class.

karelhala reviewed (approved):
Looking good!

kgaikwad commented:
Thank you @karelhala for review :ok_hand: :+1:
